### ISSUE TITLE
Change in MultiBinding behaviour in 11.1.0-rc1 compared to 11.0.11

### ISSUE THREAD

**Issue by mysticpillow** [AvaloniaUI/Avalonia#16084]
### Describe the bug

Manually calling `ObservableObject.OnPropertyChanged(propertyName)` does not trigger MultiBinding in 11.1.0-rc1 compared to 11.0.11.
It still works after scrolling down to hide node from view and up to show again. Also works for child nodes after collapsing/expanding parent node.

### To Reproduce

**XAML code inside `DataTemplate` of `TreeDataGrid` column**
```
<ContentControl.Content>
    <MultiBinding Converter="{x:Static converters:DataUseStateToIconConverter.Instance}">
        <Binding DataType="views:CustomTree" Path="#customTree" />
        <Binding DataType="models:TreeNode" Path="(models:TreeNode)" />
        <Binding DataType="models:TreeNode" Path="(models:TreeNode).DataUseStateUpdater"/>
    </MultiBinding>
</ContentControl.Content>
```
**Code-behnid**
`OnPropertyChanged(nameof(DataUseStateUpdater));` is called inside `TreeNode` model
`TreeNode` is `ObservableObject`
`DataUseStateUpdater` is `ObservableProperty`

### Expected behavior

Calling `OnPropertyChanged(propertyName)` manually should trigger `MultiBinding` with a `Binding` to the property. Then converter should be used.

### Avalonia version

11.1.0-rc1

### OS

Windows

**Comment by timunie** [comment]
Please file a minimum sample to reproduce.

**Comment by mysticpillow** [comment]
A minimum repro: https://github.com/mysticpillow/MultiBindingRegressionRepro
It uses a simple `TextBlock` instead of `TreeDataGrid`.
Actually `MultiBinding` stopped working at 11.1.0-beta1 version.

**Comment by mysticpillow** [comment]
I think #16137 is closely related to this issue. In my example a `MultiBinding` converter (which is also custom) is used only once and then ignores following `OnPropertyChanged` calls in the same way.

**Comment by grokys** [comment]
Yep, this looks like the same issue as #16137 in that you're raising a `PropertyChanged` event on a property that hasn't changed, and expecting that to trigger a side-effect. My comment in https://github.com/AvaloniaUI/Avalonia/issues/16137#issuecomment-2192466488 also stands for this issue.

Given that these two issues have the same root cause I'm going to close this issue and track it in #16137.

**Comment by grokys** [comment]
Reopening this issue because although it has the same root cause as #16137, the fix for it is more difficult. I don't think a fix for this regression will make it into 11.1.0 as:

- The fix is more involved, and potentially destabilizing
- It's even more of an edge-case than #16137

**Comment by timunie** [comment]
@spacilv should be available in nightly or 11.2 as per https://github.com/AvaloniaUI/Avalonia/issues/16084#issuecomment-2196468378
